BenchPress/Helpers/BenchPress.Azure/Authentication.psm1

```diff
@@ -0,0 +1,164 @@
+#>
+function Get-RequiredEnvironmentVariable {
+  [OutputType([System.String])]
+  param (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string]$VariableName
+  )
+  Begin {
+    $Value = [string]$null
+  }
+  Process {
+    $Value = [System.Environment]::GetEnvironmentVariable($VariableName)
+
+    if ([string]::IsNullOrWhiteSpace($Value)) {
+      Write-Error("Missing Required Environment Variable $VariableName")
+      exit 1
+    }
+  }
+  End {
+    return $Value
+  }
+}
+
+<#
+  .SYNOPSIS
+    Connect-Account uses environment variable values to log into an Azure context. This is an internal function and
+    should not be used outside of the module.
+
+  .DESCRIPTION
+    Connect-Account is designed to login to an Azure context using environment variables to login as a
+    ServicePrincipal for the PowerShell session.
+
+    The expected environment variables are:
+
+    AZ_APPLICATION_ID - The Service Principal ID
+    AZ_ENCRYPTED_PASSWORD - The Service Principal account password properly encrypted using ConvertTo-SecureString and
+                            saved as an environment variable using ConvertFrom-SecureString
+    AZ_TENANT_ID - The Tenant ID to login to
+    AZ_SUBSCRIPTION_ID - The Subscription ID to login to
+
+    If the current context that is logged in to matches the Service Principal, Tenant, and Subscription this function
+    is a no-op.
+
+  .EXAMPLE
+    There is only one way to call Connect-Account:
+
+    Connect-Account
+
+  .INPUTS
+    None
+
+  .OUTPUTS
+    None
+#>
+function Connect-Account {
+  [OutputType([System.Void])]
+  param ( )
+  Begin {
+    $ApplicationId = Get-RequiredEnvironmentVariable AZ_APPLICATION_ID
+    $TenantId = Get-RequiredEnvironmentVariable AZ_TENANT_ID
+    $SubscriptionId = Get-RequiredEnvironmentVariable AZ_SUBSCRIPTION_ID
+    $CurrentConnection = Get-AzContext
+  }
+  Process {
+    # If the current context matches the subscription, tentant, and service principal, then we're already properly
+    # logged in.
+    if ($null -ne $CurrentConnection `
+      -and ($CurrentConnection).Account.Type -eq 'ServicePrincipal' `
+      -and ($CurrentConnection).Account.Id -eq $ApplicationId `
+      -and ($CurrentConnection).Tenant.Id -eq $TenantId `
+      -and ($CurrentConnection).Subscription.Id -eq $SubscriptionId) {
+      return
+    }
+
+    # The current context is not correct, create the credentials and login
+    $ClientSecret = Get-RequiredEnvironmentVariable AZ_ENCRYPTED_PASSWORD | ConvertTo-SecureString
+    $Credential = `
+      New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $ApplicationId, $ClientSecret
+    Connect-AzAccount -ServicePrincipal -Credential $Credential -TenantId $TenantId -Subscription $SubscriptionId
+  }
+  End { }
+}
+
+<#
+  .SYNOPSIS
+    Disconnect-Account uses environment variable values to disconnect from a specific Azure context. This is an
+    internal function and should not be used outside of the module.
```

Comment:
NIT:
```suggestion
    internal function and should not be used outside of an internal module.
```

Same for the other file descriptions (to reduce confusion as to if this is in only in this file or general internal modules)